spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -620,20 +626,23 @@ private static class WriterFactory implements DataWriterFactory, StreamingDataWr
     private final Schema writeSchema;
     private final StructType dsSchema;
     private final boolean partitionedFanoutEnabled;
+    private final boolean isOrderedWrite;
```

Comment:
@ajantha-bhat, I think this should pass the sort order ID that was used instead. There isn't a guarantee that the write used the default sort order. It could be possible to choose the sort order in the future, and some of the actions already allow you to choose.

Reply:
This `isOrderedWrite` is decided based on `requiredOrdering` field of `SparkWrite` which is an Array of Spark SortOrder. I don't know how to convert array of spark sort order to an Iceberg sort order (didn't find an existing code). I guess only Iceberg sort order has the sort order ID.